py/torch_tensorrt/dynamo/conversion/_TRTInterpreter.py

```diff
@@ -393,8 +413,19 @@ def check_weight_equal(
             )
         }
         """
+        _LOGGER.info("building weight name mapping...")
         # Stage 1: Name mapping
         sd = self.module.state_dict()
+        torch_device = to_torch_device(self.compilation_settings.device)
+        gm_is_on_cuda = list(sd.values())[0].device.type == "cuda"
+        if not gm_is_on_cuda:
+            # If the model original position is on CPU, move it GPU
+            sd = {
+                k: v.reshape(-1).to(torch_device)
+                for k, v in self.module.state_dict().items()
+            }
```

Comment:
is there a necessity for the model to be on cuda for saving the weight mapping ?

Reply:
We want to use cuda to accelerate the weight-finding process, especially when checking whether two weights are equal. This only occupies 1x model size and will not cause extra space.

Reply:
I think its okay to use GPU for computation throughout the compilation process but we just want to clear memory before building occurs

Reply:
Yes I destroyed at the end of this function